### ISSUE TITLE
[1.2.1] unnecessary setter exception

### ISSUE THREAD

**Issue by musketyr** [CodeNarc/CodeNarc#364]
after upgrading to `1.2.1` some `UnecessarySetter` rules are throwing errors:

```
Error from [org.codenarc.rule.unnecessary.UnnecessarySetterRule] processing source file [path/to/some/Clazz.groovy]
java.lang.StringIndexOutOfBoundsException: String index out of range: 5
        at java.lang.String.substring(String.java:1950)
        at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1528)
        at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1515)
        at org.codehaus.groovy.runtime.dgm$1137.invoke(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:246)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:55)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.codenarc.rule.unnecessary.UnnecessarySetterAstVisitor.addViolationsIfSetter(UnnecessarySetterRule.groovy:69)
        at org.codenarc.rule.unnecessary.UnnecessarySetterAstVisitor.this$5$addViolationsIfSetter(UnnecessarySetterRule.groovy)
        at sun.reflect.GeneratedMethodAccessor1162.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:169)
        at org.codenarc.rule.unnecessary.UnnecessarySetterAstVisitor.visitExpressionStatement(UnnecessarySetterRule.groovy:42)
        at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:42)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:88)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:106)
        at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitIfElse(CodeVisitorSupport.java:109)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitIfElse(ClassCodeVisitorSupport.java:207)
        at org.codehaus.groovy.ast.stmt.IfStatement.visit(IfStatement.java:43)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:88)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:106)
        at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitIfElse(CodeVisitorSupport.java:109)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitIfElse(ClassCodeVisitorSupport.java:207)
        at org.codehaus.groovy.ast.stmt.IfStatement.visit(IfStatement.java:43)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:88)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:106)
        at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitForLoop(CodeVisitorSupport.java:94)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitForLoop(ClassCodeVisitorSupport.java:202)
        at org.codehaus.groovy.ast.stmt.ForStatement.visit(ForStatement.java:49)
        at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:88)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:106)
        at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:110)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:121)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:132)
        at org.codenarc.rule.AbstractAstVisitor.visitMethod(AbstractAstVisitor.java:155)
        at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1103)
        at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:54)
        at org.codenarc.rule.AbstractAstVisitor.visitClass(AbstractAstVisitor.java:132)
        at org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
        at org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:143)
        at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
        at org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
        at org.codenarc.ant.AntFileSetSourceAnalyzer.processFile(AntFileSetSourceAnalyzer.java:171)
        at org.codenarc.ant.AntFileSetSourceAnalyzer.access$000(AntFileSetSourceAnalyzer.java:42)
        at org.codenarc.ant.AntFileSetSourceAnalyzer$1.run(AntFileSetSourceAnalyzer.java:160)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:744)
```

**Comment by chrismair** [comment]
Are you able to provide some Groovy code that is causing the issue?

**Comment by chrismair** [comment]
Found it. Code like this causes that exception:
```
            x.setE(3)
```
In other words, a method name named "set" + one capital letter.

**Comment by chrismair** [comment]
Fixed for CodeNarc 1.3
